weblab/entities/views.py

```diff
@@ -123,6 +123,36 @@ def get_commit(self):
 
         return self._commit
 
+    def get_version(self):
+        """
+        Get the cached entity for this  version
+
+        :return: CachedEntityVersion object
+        :raise: Http404 if version not found
+        """
+        sha_or_tag = self.kwargs['sha']
```

Comment:
Yes I have been wrestling with this question of `_version`. I can see that here it is useful to protect against "requerying" when there is already a version. `_commit` comes into play because of the `__eq__ `override in Commit class. 

I have changed to using _version in get_version but haven't implemented the equality class (in CachedEntityVersion) and all the test pass. So hopefully this is going to be ok?

Reply:
Django has a default implementation of `__eq__` for DB model subclasses which checks that the IDs match. Which would do the right thing for us here, so should be OK.